altair/utils/schemapi.py

```diff
@@ -873,12 +915,19 @@ def to_json(
         return json.dumps(dct, indent=indent, sort_keys=sort_keys, **kwargs)
 
     @classmethod
-    def _default_wrapper_classes(cls):
+    def _default_wrapper_classes(cls) -> Generator[Type["SchemaBase"], None, None]:
         """Return the set of classes used within cls.from_dict()"""
         return _subclasses(SchemaBase)
 
     @classmethod
-    def from_dict(cls, dct, validate=True, _wrapper_classes=None):
+    def from_dict(
+        cls,
+        dct: dict,
+        validate: bool = True,
+        _wrapper_classes: Optional[Iterable[Type["SchemaBase"]]] = None,
+        # Type hints for this method would get rather complicated
+        # if we want to provide a more specific return type
+    ) -> Any:
```

Comment:
Would `_TSchemaBase` work here? I seems like the returned object should be a `SchemaBase` subclass.

Reply:
When I used `SchemaBase` as a return type, I got an error for `Chart.from_dict` in `api.py` so I left it at that. But after your comment I revisited this and indeed the `Chart.from_dict` type was wrong :) -> Done in 2732c8f402ea594191168ee89